pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -88,28 +88,24 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 		return fmt.Errorf("spec.externalEtcdConfiguration is immutable")
 	}
 
-	if !v1alpha1.RefSliceEqual(nSpec.IdentityProviderRefs, oSpec.IdentityProviderRefs) {
-		return fmt.Errorf("spec.identityProviderRefs is immutable")
+	oldAWSIamConfigRef := &v1alpha1.Ref{}
+
+	for _, oIdentityProvider := range oSpec.IdentityProviderRefs {
+		switch oIdentityProvider.Kind {
+		case v1alpha1.AWSIamConfigKind:
+			oldAWSIamConfigRef = &oIdentityProvider
+		}
 	}
-	if len(nSpec.IdentityProviderRefs) > 0 {
-		for _, nIdentityProvider := range nSpec.IdentityProviderRefs {
-			switch nIdentityProvider.Kind {
-			case v1alpha1.OIDCConfigKind:
-				prevOIDC, err := k.GetEksaOIDCConfig(ctx, nIdentityProvider.Name, cluster.KubeconfigFile, spec.Namespace)
-				if err != nil {
-					return err
-				}
-				if !prevOIDC.Spec.Equal(&spec.OIDCConfig.Spec) {
-					return fmt.Errorf("oidc identity provider is immutable")
-				}
-			case v1alpha1.AWSIamConfigKind:
-				prevAwsIam, err := k.GetEksaAWSIamConfig(ctx, nIdentityProvider.Name, cluster.KubeconfigFile, spec.Namespace)
-				if err != nil {
-					return err
-				}
-				if !prevAwsIam.Spec.Equal(&spec.AWSIamConfig.Spec) {
-					return fmt.Errorf("aws iam identity provider is immutable")
-				}
+	for _, nIdentityProvider := range nSpec.IdentityProviderRefs {
+		switch nIdentityProvider.Kind {
+		case v1alpha1.AWSIamConfigKind:
+			newAWSIamConfigRef := &nIdentityProvider
+			prevAwsIam, err := k.GetEksaAWSIamConfig(ctx, nIdentityProvider.Name, cluster.KubeconfigFile, spec.Namespace)
+			if err != nil {
+				return fmt.Errorf("aws iam identity provider is immutable: %v", err)
```

Comment:
Can we just return err here? 
k.GetEksaAWSIamConfig already adds some context to the error

Reply:
In this case if AWS isn't configured it only returns saying no awsiamconfig object found, so I added the context saying it's immutable.